### PR TITLE
API docs

### PR DESCRIPTION
**Pull request by 10xLaCroixDrinker**
- Added API section to `holocron-module-route` README
- Structure change in `holocron`'s API.md
- Added API section to `iguazu-holocron` README

### CODE REVIEW

File: packages/iguazu-holocron/README.md [hunk 1]
```diff
@@ -120,6 +121,176 @@ const loadDataAsProps = ({ store: { dispatch } }) => ({
...
+
+MyModule.loadDataAsProps = loadDataAsProps;
+
+if (!global.BROWSER) {
+  MyModule.loadModuleData = configureIguazuSSR;
+}
+
+export default connectAsync({ loadDataAsProps })(MyModule);
+```
+
```
Comment (mtomcal) [inline packages/iguazu-holocron/README.md]: Should have `More Information` section for referencing `loadModuleData` Module Lifecycle Hook. Also we dont need `if (!global.BROWSER)` for this function as it is universal. The if-check is for `appConfig` only.

File: packages/iguazu-holocron/README.md [hunk 2]
```diff
@@ -10,6 +10,7 @@
 ## 📖 Table of Contents
```
Comment (mtomcal) [inline packages/iguazu-holocron/README.md]: In the description can we add a link to Iguazu?

File: packages/holocron/API.md [hunk 3]
```diff
@@ -168,105 +205,114 @@ const load = () => (dispatch) => dispatch(loadModule('my-module'));
...
 import { setModuleMap } from 'holocron';
 
 setModuleMap(newModuleMap);
 ```
 
-### `isLoaded`
+### Selectors
+
+These are all functions that take a module name as the only parameter and return a new function
+that accepts the [Redux] state as the only parameter which returns data about the module.
```
[the remark this answers is not in the document]
Comment (10xLaCroixDrinker) [reply packages/holocron/API.md]: This is a link to redux

File: packages/holocron-module-route/README.md [hunk 4]
```diff
@@ -70,15 +76,166 @@ export default function getChildRoutes(store) {
...
+`ModuleRoute` extends [`one-app-router`](https://github.com/americanexpress/one-app-router)'s 
+[`Route`](https://github.com/americanexpress/one-app-router/blob/master/docs/API.md#route). It has 
+all the functionality of `Route`, with the addition of loading a holocron module instead of a 
+component.
+
+#### Props
+
+| name | type | required | value |
+|---|---|---|---|
+| `moduleName` | `String` | `false` | The name of the Holocron module to render when the route is matched |
```
[the remark this answers is not in the document]
Comment (10xLaCroixDrinker) [reply packages/holocron-module-route/README.md]: I dont want to include any that arent specific to ModuleRoute

File: packages/holocron-module-route/README.md [hunk 5]
```diff
@@ -70,15 +76,166 @@ export default function getChildRoutes(store) {
...
+
+MyModule.onEnterRouteHook = (store) => (nextState, replace, callback) => {
+  store.dispatch(authenticate()).then(() => {
+    if (!store.getState().isAuthenticated) replace('/login');
+    callback();
+  });
+}
+```
+
+### `moduleRoutePrefetch`
```
[the remark this answers is not in the document]
Comment (10xLaCroixDrinker) [reply packages/holocron-module-route/README.md]: This isnt under lifecycle hook int the hierachy, it is a sibling to `ModuleRoute`

File: packages/holocron-module-route/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (mtomcal) [reply packages/holocron-module-route/README.md]: 👍

File: packages/holocron/API.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (mtomcal) [reply packages/holocron/API.md]: Maybe the diff lied to me:
```that accepts the [Redux] state as the only parameter which returns data about the module.``` is what I see

File: packages/holocron-module-route/README.md [hunk 6]
```diff
@@ -70,15 +76,166 @@ export default function getChildRoutes(store) {
 }
 ```
 
+## 🎛️ API
```
Comment (mtomcal) [inline packages/holocron-module-route/README.md]: A table is broken: https://github.com/americanexpress/holocron/blob/6efd12bf72721be99bc5124fb01f11c2a0e34c72/packages/holocron-module-route/README.md#module-lifecycle-hooks

File: packages/holocron/API.md [hunk 7]
```diff
@@ -16,7 +48,7 @@ Creates the [Redux] store with Holocron compatibility.
 | `localsForBuildInitialState` | `Object` | `false` | Value to pass to [vitruvius]'s `buildInitialState` |
 | `extraThunkArguments` | `Object` | `false` | Additional arguments to be passed to [Redux thunks] |
```
Comment (mtomcal) [inline packages/holocron/API.md]: `Additional arguments to be passed to [Redux thunks]` No url?

File: packages/holocron/API.md [hunk 8]
```diff
@@ -93,19 +130,19 @@ Components using this HOC will be provided several props.
 | `moduleLoadStatus` | `String` | One of `"loading"`, `"loaded"`, or `"error"`, based on the `load` function |
 | `moduleState` | `Object` | The state of the registered reducer after [`.toJS()`] has been called on it |
```
Comment (mtomcal) [inline packages/holocron/API.md]: URL? ```[`.toJS()`]```

**Comment by 10xLaCroixDrinker** [comment]
> I think `[blah]` syntax doesnt work in github today.

The issue was how ther references were created at the bottom. All fixed now

**Pull request merged**